packages/accounts/src/kernel-zerodev/provider.ts

```diff
@@ -0,0 +1,54 @@
+import {Address, Chain, Transport} from "viem";
+import {KernelSmartContractAccount} from "./account";
+import {defineReadOnly} from "@alchemy/aa-core/src/utils";
+import {
+    PublicErc4337Client,
+    SmartAccountProvider,
+    SmartAccountProviderOpts,
+    SupportedTransports
+} from "@alchemy/aa-core/src";
+
+export class KernelAccountProvider<
+    TTransport extends SupportedTransports = Transport
+> extends SmartAccountProvider<TTransport> {
+
+
+    constructor(
+        rpcProvider: string | PublicErc4337Client<TTransport>,
+        private entryPointAddress: Address,
+        private chain: Chain,
+        readonly account?: KernelSmartContractAccount,
+        opts?: SmartAccountProviderOpts
+    ) {
+        super(rpcProvider,entryPointAddress,chain,account,opts)
+    }
```

Comment:
is this constructor necessary? the `entryPointAddress` and `chain` are now protected properties. Only thing I can see that's specific is the requirement for `KernelSmartContractAccount` on the account type.

we could update the interface for `ISmartAccountProvider` and the base provider to accept one more generic:
```ts
interface ISmartAccountProvider<
  TTransport extends SupportedTransports = Transport,
  TSmartContractAccount extends BaseSmartContractAccount = BaseSmartContractAccount,
>
```
and then here you would have 

```ts
export class KernelAccountProvider<
    TTransport extends SupportedTransports = Transport
> extends SmartAccountProvider<TTransport, KernelSmartContractAccount>
```

this would reduce the boilerplate for future contributors

Reply:
Noted. will do

Reply:
https://github.com/alchemyplatform/aa-sdk/pull/28/files